### DISCUSSION TITLE
Add Viz framework support

### DISCUSSION THREAD

**Discussion by fundon** [Totodore/socketioxide#198] [Q&A]
[viz-rs/viz:](https://github.com/viz-rs/viz) Fast, flexible, lightweight web framework for Rust

**Comment by fundon** [comment]
I can create a PR. :)

**Reply by Totodore** [reply]
The `viz` framework needs to support raw `hyper::service` or `tower::service` or `tower::layers` so that socketioxide is compatible. Because socketioxide can't contain specific framework code. 

Here is an equivalent PR on the salvo repo : https://github.com/salvo-rs/salvo/pull/435

If you really want socketioxide to support viz and that viz doesn't want to support tower layers/services. You can make an external adapter lib called `viz-tower` for instance that will adapt a tower layer / service to a viz `Handler`.

**Reply by Totodore** [reply]
Once you have a working example don't hesitate to do a PR to add a `viz-echo` example as well as any other you want
